### ISSUE TITLE
[WriteSupport] Write a Dataframe to Neo4j as nodes

### ISSUE THREAD

**Issue by conker84** [neo4j-contrib/neo4j-spark-connector#144]
We want to save a Dataframe as Nodes
```scala
otherFormatDf
   .write
   .format("neo4j")
   .mode(SaveMode.Overwrite)
   .option("node", "Person")
   .option("nodeKeys", "ssn") // comma separated list of keys
   .save()
```

For the given Dataframe:

ssn | name | surname
-- | -- | --
AB12 | Andrea | Santurbano
CD34 | Davide | Fantuzzi

Under-the-hood we'll perform the following query:

```cypher
UNWIND batch AS row
MERGE (p:Person{ssn: row.ssn}) SET p += row
```

We'll support two `SaveMode` (https://spark.apache.org/docs/2.4.4/api/java/index.html?org/apache/spark/sql/SaveMode.html):
* `Overwrite`: this will perform a `MERGE` query
* `ErrorIfExists`: this will perform a `CREATE` query

**Comment by moxious** [comment]
* nodeKeys can be an array (unlikely but possible)
* nodeKeys is optional if you're using CREATE
* See other comments on write in general, batchSize and failureMode.

**Comment by conker84** [comment]
closed via 1c1abc8
